app/src/main/java/com/keylesspalace/tusky/components/compose/view/ComposeScheduleView.kt

```diff
@@ -0,0 +1,227 @@
+/* Copyright 2019 kyori19
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+package com.keylesspalace.tusky.components.compose.view
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import androidx.appcompat.app.AppCompatActivity
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.content.ContextCompat
+import com.google.android.material.datepicker.CalendarConstraints
+import com.google.android.material.datepicker.DateValidatorPointForward
+import com.google.android.material.datepicker.MaterialDatePicker
+import com.google.android.material.timepicker.MaterialTimePicker
+import com.google.android.material.timepicker.TimeFormat
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.databinding.ViewComposeScheduleBinding
+import java.text.DateFormat
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Date
+import java.util.Locale
+import java.util.TimeZone
+
+class ComposeScheduleView : ConstraintLayout {
+    interface OnTimeSetListener {
+        fun onTimeSet(time: String?)
+    }
+
+    private lateinit var binding: ViewComposeScheduleBinding
+    private var listener: OnTimeSetListener? = null
+    private lateinit var dateFormat: DateFormat
+    private lateinit var timeFormat: DateFormat
+    private lateinit var iso8601: SimpleDateFormat
+    private var scheduleDateTime: Calendar? = null
+
+    constructor(context: Context) : super(context) {
+        init(context)
+    }
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
+        init(context)
+    }
+
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
+        context,
+        attrs,
+        defStyleAttr
+    ) {
+        init(context)
+    }
```

Comment:
please merge the constructors into one with default values like e.g. https://github.com/tuskyapp/Tusky/blob/develop/app/src/main/java/com/keylesspalace/tusky/components/compose/view/FocusIndicatorView.kt#L23

Reply:
also in ProgressImageView

Reply:
Ah, neat trick.